File: app/src/main/java/com/example/delhitransit/Data/DataClasses/BusPositionUpdate.java
package com.example.delhitransit.Data.DataClasses;

import androidx.annotation.NonNull;
import androidx.room.Entity;
import androidx.room.Ignore;
import androidx.room.PrimaryKey;

import com.example.delhitransit.GtfsRealtime;

@Entity(tableName = "positions")
public class BusPositionUpdate {

    @PrimaryKey
    @NonNull
    private String vehicleID = "";

    private String tripID;
    private String routeID;
    private float latitude;
    private float longitude;
    private float speed;
    private long timestamp;

    public BusPositionUpdate(String vehicleID, String tripID, String routeID, float latitude, float longitude, float speed, long timestamp) {
        this.vehicleID = vehicleID;
        this.tripID = tripID;
        this.routeID = routeID;
        this.latitude = latitude;
        this.longitude = longitude;
        this.speed = speed;
        this.timestamp = timestamp;
    }

    @Ignore
    public BusPositionUpdate() {
    }

    @Ignore
    public BusPositionUpdate parseFrom(GtfsRealtime.FeedEntity entity) {
        this.vehicleID = entity.getId();

        GtfsRealtime.VehiclePosition vehicle = entity.getVehicle();
        GtfsRealtime.TripDescriptor trip = vehicle.getTrip();
        GtfsRealtime.Position position = vehicle.getPosition();

        this.tripID = trip.getTripId();
        this.routeID = trip.getRouteId();

        this.latitude = position.getLatitude();
        this.longitude = position.getLongitude();
        this.speed = position.getSpeed();

        this.timestamp = vehicle.getTimestamp();

        return this;
    }

    public String getVehicleID() {
        return vehicleID;
    }

    public String getTripID() {
        return tripID;
    }

    public String getRouteID() {
        return routeID;
    }

    public float getLatitude() {
        return latitude;
    }

    public float getLongitude() {
        return longitude;
    }

    public float getSpeed() {
        return speed;
    }

    public long getTimestamp() {
        return timestamp;
    }

    @Override
    public String toString() {
        StringBuilder builder = new StringBuilder();
        builder.append("\nVehicle ID:\t");
        builder.append(vehicleID);
        builder.append("\nTrip ID:\t");
        builder.append(tripID);
        builder.append("\nRoute ID:\t");
        builder.append(routeID);
        builder.append("\nBus Coordinates:\t");
        builder.append(latitude);
        builder.append(",");
        builder.append(longitude);
        builder.append("\nSpeed:\t");
        builder.append(speed);
        builder.append("\nTimestamp:\t");
        builder.append(timestamp);
        return builder.toString();
    }
}

File: app/src/main/java/com/example/delhitransit/BusPositionUpdate.kt
package com.example.delhitransit

data class BusPositionUpdate(var vehicleID : String = "",var tripID: String? = null,var routeID: String? = null,var latitude: Float = 0.toFloat(),var longitude: Float = 0.toFloat(),var speed: Float = 0.toFloat(),var timestamp: Long = 0){

    fun parseFrom(entity: GtfsRealtime.FeedEntity): BusPositionUpdate {
        this.vehicleID = entity.id

        val vehicle = entity.vehicle
        val trip = vehicle.trip
        val position = vehicle.position

        this.tripID = trip.tripId
        this.routeID = trip.routeId

        this.latitude = position.latitude
        this.longitude = position.longitude
        this.speed = position.speed

        this.timestamp = vehicle.timestamp

        return this
    }

    override fun toString(): String {
        val builder = StringBuilder()
        builder.append("\nVehicle ID:\t")
        builder.append(vehicleID)
        builder.append("\nTrip ID:\t")
        builder.append(tripID)
        builder.append("\nRoute ID:\t")
        builder.append(routeID)
        builder.append("\nBus Coordinates:\t")
        builder.append(latitude)
        builder.append(",")
        builder.append(longitude)
        builder.append("\nSpeed:\t")
        builder.append(speed)
        builder.append("\nTimestamp:\t")
        builder.append(timestamp)
        return builder.toString()
    }
}
